gobblin-core/src/main/java/gobblin/http/DefaultHttpClientConfigurator.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.http;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.http.HttpHost;
+import org.apache.http.client.HttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Optional;
+import com.google.common.base.Strings;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigValueFactory;
+
+import gobblin.annotation.Alias;
+import gobblin.configuration.State;
+
+/**
+ * Default implementation that uses the following properties to configure an {@link HttpClient}.
+ *
+ * <ul>
+ *  <li>{@link #PROXY_HOSTPORT_KEY}
+ *  <li>{@link #PROXY_URL_KEY}
+ *  <li>{@link #PROXY_PORT_KEY}
+ * </ul>
+ */
+@Alias(value="default")
+public class DefaultHttpClientConfigurator implements HttpClientConfigurator {
+  // IMPORTANT: don't change the values for PROXY_URL_KEY and PROXY_PORT_KEY as they are meant to
+  // be backwards compatible with SOURCE_CONN_USE_PROXY_URL and SOURCE_CONN_USE_PROXY_PORT when
+  // the statePropertiesPrefix is "source.conn."
+  /** The hostname of the HTTP proxy to use */
+  public static final String PROXY_URL_KEY = "use.proxy.url";
+  /** The port of the HTTP proxy to use */
+  public static final String PROXY_PORT_KEY = "use.proxy.port";
+  /** Similar to {@link #PROXY_URL_KEY} and {@link #PROXY_PORT_KEY} but allows you to set it on
+   * one property as <host>:<port> */
+  public static final String PROXY_HOSTPORT_KEY = "proxyHostport";
```

Comment:
Shall we mark it explicitly that setting of `PROXY_HOSTPORT_KEY` will override other two keys if they are all used?